pkg/reconciler/gc/controller.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+	"time"
+
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	configurationinformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/configuration"
+	revisioninformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/revision"
+	"knative.dev/serving/pkg/reconciler"
+	configns "knative.dev/serving/pkg/reconciler/configuration/config"
+)
+
+const (
+	controllerAgentName = "gc-controller"
+	resyncInterval      = 4 * time.Hour
+)
+
+// NewController creates a new Garbage Collection controller
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+
+	configurationInformer := configurationinformer.Get(ctx)
```

Comment:
You should use https://github.com/knative/serving/blob/4f4739b0cb8838133efa847fedc061480f16527b/vendor/knative.dev/pkg/controller/controller.go#L438 to get actually resync behaviour you're looking for.